src/cargo/ops/cargo_update.rs

```diff
@@ -207,6 +221,215 @@ pub fn print_lockfile_changes(
+                            } else {
+                                None
+                            };
+
+                            if let Some(latest) = latest {
+                                if !current.matches(&latest) {
+                                    trace!(
+                                        "upgrading `{}` from {} to {}",
+                                        name,
+                                        current,
+                                        latest
+                                    );
+
+                                    opts.gctx.shell().status_with_color(
+                                        "Upgrading",
+                                        format!(
+                                            "{} {} -> v{}",
+                                            name,
+                                            current,
+                                            latest.to_string()
+                                        ),
+                                        &style::GOOD,
+                                    )?;
+
+                                    upgrades.insert(
+                                        name.to_string(),
+                                        latest.clone(),
+                                    );
+
+                                    let req =
+                                        OptVersionReq::Req(VersionReq::parse(&latest.to_string())?);
+                                    let mut dep = dependency.clone();
+                                    dep.set_version_req(req);
+                                    return Ok(dep);
+                                } else {
+                                    trace!(
+                                        "skipping dependency `{}` without a breaking update available",
+                                        name
+                                    );
+                                }
+                                            } else {
+                                trace!(
+                                    "skipping dependency `{}` without any published versions",
+                                    name
+                                );
+                            }
+                        } else {
+                            trace!("skipping non-caret dependency `{}`: {}", name, comparator);
+                        }
+                    } else {
+                        trace!(
+                            "skipping dependency `{}` with multiple version comparators: {:?}",
+                            name,
+                            &current.comparators
+                        );
+                    }
+                } else {
+                    trace!(
+                        "skipping dependency `{}` without a simple version requirement: {}",
+                        name,
+                        version_req
+                    );
+                }
```

Comment:
Would there be a good way to shift to early returns / breaks rather than this deep nesting for trivial branches?

I find code like this tends to be harder to read and I generally recommend nesting logic to be focused on business cases while using early returns for bookkeeping makes things a lot clearer.

Reply:
Agreed. Fixed.